packages/components/src/font-size-picker/index.js

```diff
@@ -26,76 +31,61 @@ function FontSizePicker( {
 } ) {
+	// eslint-disable-next-line @wordpress/no-unused-vars-before-return
+	const [ currentSelectValue, setCurrentSelectValue ] = useState( getInitialSelectValue( fontSizes, value ) );
+
 	if ( disableCustomFontSizes && ! fontSizes.length ) {
 		return null;
 	}
+
 	const onChangeValue = ( event ) => {
 		const newValue = event.target.value;
 		if ( newValue === '' ) {
 			onChange( undefined );
 			return;
 		}
+		setCurrentSelectValue( getInitialSelectValue( fontSizes, Number( newValue ) ) );
 		onChange( Number( newValue ) );
 	};
 
-	const currentFont = fontSizes.find( ( font ) => font.size === value );
-	const currentFontSizeName = ( currentFont && currentFont.name ) || ( ! value && _x( 'Normal', 'font size name' ) ) || _x( 'Custom', 'font size name' );
+	const onSelectChangeValue = ( eventValue ) => {
+		setCurrentSelectValue( eventValue );
+		const selectedFont = fontSizes.find( ( font ) => font.slug === eventValue );
+
+		if ( selectedFont ) {
+			onChange( selectedFont.size );
+		}
+	};
 
 	return (
-		<BaseControl>
-			<BaseControl.VisualLabel>
+		<fieldset>
+			<legend>
 				{ __( 'Font Size' ) }
-			</BaseControl.VisualLabel>
-			<div className="components-font-size-picker__buttons">
+			</legend>
+			<div className="components-font-size-picker__controls">
 				{ ( fontSizes.length > 0 ) &&
-					<Dropdown
-						className="components-font-size-picker__dropdown"
-						contentClassName="components-font-size-picker__dropdown-content"
-						position="bottom"
-						renderToggle={ ( { isOpen, onToggle } ) => (
-							<Button
-								className="components-font-size-picker__selector"
-								isLarge
-								onClick={ onToggle }
-								aria-expanded={ isOpen }
-								aria-label={ sprintf(
-									/* translators: %s: font size name */
-									__( 'Font size: %s' ), currentFontSizeName
-								) }
-							>
-								{ currentFontSizeName }
-							</Button>
-						) }
-						renderContent={ () => (
-							<NavigableMenu>
-								{ map( fontSizes, ( { name, size, slug } ) => {
-									const isSelected = ( value === size || ( ! value && slug === 'normal' ) );
-
-									return (
-										<Button
-											key={ slug }
-											onClick={ () => onChange( slug === 'normal' ? undefined : size ) }
-											className={ `is-font-${ slug }` }
-											role="menuitemradio"
-											aria-checked={ isSelected }
-										>
-											{ isSelected && <Dashicon icon="saved" /> }
-											<span className="components-font-size-picker__dropdown-text-size" style={ { fontSize: size } }>
-												{ name }
-											</span>
-										</Button>
-									);
-								} ) }
-							</NavigableMenu>
-						) }
+					<SelectControl
+						className={ 'components-font-size-picker__select' }
+						label={ 'Choose preset' }
+						hideLabel={ true }
+						value={ currentSelectValue }
+						onChange={ onSelectChangeValue }
+						options={ [
+							{ value: fontSizes[ 0 ].slug, label: __( 'Small' ) },
```

Comment:
Font sizes can be customized for each site: 
https://developer.wordpress.org/block-editor/developers/themes/theme-support/#block-font-sizes

The names used as labels and the number of items may differ.

Reply:
I tried to generate `options` dynamically from the `fontSizes` array but it seems the i18n `__` function can't be called with dynamic content. Is there any workaround for this?

Reply:
When those options are set on the server, they should be already translated. There is no need to wrap them with `__` on the client.

Reply:
Got it! Thanks.